### ISSUE TITLE
AbstractFileReference>>/ with empty string as argument answers filesystem root

### ISSUE THREAD

**Issue by theseion** [guillep/pharo-core#6182]
**Describe the bug**
AbstractFileReference>>/ with empty string as argument answers filesystem root.

**To Reproduce**
Evaluate `FileLocator workingDirectory / ''`

**Expected behavior**
Either an error should be signalled or the empty filename should just be ignored.
Actually, an error is signalled but not the one I want. In File>>openForWrite: `CannotDeleteFileException` is signalled because no file handle could be obtained. While that is correct it is not reflective of the actual problem.

**Version information:**
- OS: macOS 10.14
 - Version: Pharo 8

**Expected development cost**
Depends on how we want to fix it. Simple and dirty: maybe an hour. Making a fix that involves understanding the effects that path resolution has: possibly several days.

**Comment by astares** [comment]
@theseion It does not answer the root for me but the workingDirectory again (on **Windows**, with latest Pharo 9). This is also what I would expect.

Maybe @akgrant43 can double check

**Comment by theseion** [comment]
Interesting. I just checked on macOS with Pharo 9 and I still get root. I'm pretty sure the cause is that on Windows the empty string can not be the root of the file system, which is why an instance of `RelativePath` is answered while on macOS the empty string will be treated as an absolute path to root.

**Comment by astares** [comment]
Yes - I tried on **Ubuntu** and it also shows the effect described by @theseion similar to **MacOS**. 

![image](https://user-images.githubusercontent.com/5980033/79322506-3644c380-7f0d-11ea-8154-6d75d3a3ba7a.png)
